pkg/arvo/lib/hood/drum.hoon

```diff
@@ -329,23 +330,66 @@
   [%give %fact ~[/drum] %dill-blit !>(dill-blit)]
 ::
 ++  se-adit                                           ::  update servers
-  ^+  .
-  ::  ensure dojo connects after talk
-  =*  dojo-on-top  |=([a=* b=*] |(=(%dojo a) &(!=(%dojo b) (aor a b))))
-  %+  roll  (sort ~(tap in ray) dojo-on-top)
-  =<  .(con +>)
-  |:  $:{wel/well:gall con/_..se-adit}  ^+  con
-  =.  +>.$  con
-  =+  hig=(~(get by fur) q.wel)
-  ?:  &(?=(^ hig) |(?=(~ u.hig) =(p.wel syd.u.u.hig)))  +>.$
-  =.  +>.$  (se-text "activated app {(trip p.wel)}/{(trip q.wel)}")
-  %-  se-emit(fur (~(put by fur) q.wel ~))
+  ^+  this
+  |^
+  =/  servers=(list well:gall)
+    (sort ~(tap in ray) sort-by-priorities)
+  |-
```

Comment:
I appreciate the restyling here, the code style around these parts is... quite something[.](http://www.employmentking.co.uk/wp-content/uploads/2011/11/disgust.jpg)

Was gonna ask about the manual looping vs `+roll`, but I remember struggling with getting that to behave nicely myself. To confirm, does just `|=  [wel=well:gall _this]` not cut it for the `+roll` gate?

Reply:
yeah `roll` was being funny